diff --git a/dist/types/request.d.ts b/dist/types/request.d.ts
index 161f54cf4d5a3e7f8061b8a251cd112bde5690a6..e5a8dde175c6e7f8a7d4e2dec11a1cd201761a15 100644
--- a/dist/types/request.d.ts
+++ b/dist/types/request.d.ts
@@ -24,8 +24,8 @@ export declare class HonoRequest<P extends string = '/', I extends Input['out']
     setParams(params: ParamIndexMap | Params): void;
     param<P2 extends string = P>(key: RemoveQuestion<ParamKeys<P2>>): UndefinedIfHavingQuestion<ParamKeys<P2>>;
     param<P2 extends string = P>(): UnionToIntersection<ParamKeyToRecord<ParamKeys<P2>>>;
-    query(key: string): string | undefined;
-    query(): Record<string, string>;
+    query<T>(key: string | T): string | T | undefined;
+    query<T = Record<string, string>>(): T;
     queries(key: string): string[] | undefined;
     queries(): Record<string, string[]>;
     header(name: string): string | undefined;
